runtime/src/gpu/rocm/gpu-rocm.c

```diff
@@ -77,31 +78,20 @@ void* chpl_gpu_load_function(hipModule_t rocm_module, const char* kernel_name) {
 // this is compiler-generated
 extern const char* chpl_gpuBinary;
 
-static hipCtx_t *chpl_gpu_primary_ctx;
-
 // array indexed by device ID (we load the same module once for each GPU).
 static hipModule_t *chpl_gpu_rocm_modules;
 
 static int *deviceClockRates;
 
 
-static bool chpl_gpu_has_context() {
-  hipCtx_t rocm_context = NULL;
-
-  hipError_t ret = hipCtxGetCurrent(&rocm_context);
-
-  if (ret == hipErrorNotInitialized || ret == hipErrorDeinitialized) {
-    return false;
-  }
-  else {
-    return rocm_context != NULL;
-  }
-}
-
 static void chpl_gpu_ensure_context() {
-  hipCtx_t next_context = chpl_gpu_primary_ctx[chpl_task_getRequestedSubloc()];
+  // Some hipCtx* functions are deprecated so we're using `hipSetDevice`
+  // in its place
+  hipSetDevice(chpl_task_getRequestedSubloc());
```

Comment:
Don't this (and another call to the same function below) require ROCM_CALL?

Reply:
Good point, I've updated it to wrap the calls.